rich/progress.py

```diff
@@ -651,7 +651,7 @@ def __init__(
     def render(self, task: "Task") -> ProgressBar:
         """Gets a progress bar widget for a task."""
         return ProgressBar(
-            total=max(0, task.total),
+            total=max(0, task.total) if task.total is not None else None,
```

Comment:
I had to add loads of `if task.total is not None else` here and there, now that this "total" can be `None`